docs/src/main/asciidoc/config-custom.adoc

```diff
@@ -0,0 +1,112 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= Custom Configuration
+
+include::./attributes.adoc[]
+
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:toc:
+
+[[custom-config-source]]
+== Custom Config Sources
+
+It is possible to create a Custom Config Sources as specified in
```

Comment:
I would add a short reminder about how the config is computed (and reuse the picture from the other guide).